BarrysPeanutsTemporalJava/src/main/java/barryspeanuts/App.java

```diff
@@ -0,0 +1,127 @@
+package barryspeanuts;
+
+import barryspeanuts.model.*;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.common.RetryOptions;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class App {
+  static final String TASK_QUEUE = "BarryPeanutsTemporal";
+  static final String WORKFLOW_ID = TASK_QUEUE + "-" + UUID.randomUUID();
+  private static final Logger logger = LoggerFactory.getLogger(App.class);
+
+  @SuppressWarnings("CatchAndPrintStackTrace")
+  public static void main(String[] args) {
+    // gRPC stubs wrapper that talks to the local docker instance of temporal
+    // service.
+    WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
+    // client that can be used to start and signal workflows
+    WorkflowClient client = WorkflowClient.newInstance(service);
+
+    // worker factory that can be used to create workers for specific task queues
+    WorkerFactory factory = WorkerFactory.newInstance(client);
+
+    // Worker that listens on a task queue and hosts both workflow and activity
+    // implementations.
+    Worker worker = factory.newWorker(TASK_QUEUE);
+
+    // Workflows are stateful. So you need a type to create instances.
+    worker.registerWorkflowImplementationTypes(ShoppingCartWorkflowImpl.class);
+
+    // Start all workers created by this factory.
+    factory.start();
+    logger.info("Worker started for task queue: {} with WorkflowID : {}.", TASK_QUEUE, WORKFLOW_ID);
+
+    // now we can start running instances of our workflow - its state will be persisted
+    WorkflowOptions options =
+        WorkflowOptions.newBuilder()
+            .setTaskQueue(TASK_QUEUE)
+            .setWorkflowId(WORKFLOW_ID)
+            // set the return options
+            .setRetryOptions(
+                RetryOptions.newBuilder()
+                    .setInitialInterval(Duration.ofSeconds(1))
+                    .setMaximumInterval(Duration.ofSeconds(10))
+                    .build())
+            .build();
+
+    ShoppingCartWorkflow wf = client.newWorkflowStub(ShoppingCartWorkflow.class, options);
+    try {
+      Address address =
+          new Address("123 Main Street", "Apt 1", "Anytown", "CA", "99999-9999", "USA");
+
+      Customer customer =
+          new Customer(
+              UUID.randomUUID(),
```

Comment:
Can we move generation of ids into the data classes? That's uselessly complicating the sample.

Reply:
@mjameswh @cretz asked me not move the Ids to the data class.

Reply:
Yeah, and I understand why. I was hesitating on that one. In a real world app, the item ID would be part of the definition of the item, just like any other attributes, so having them as constructor arg would make more sense.

However, they are adding useless cognitive weight to the demo code. Plus that is exposing some incoherencies of this demo, eg. the fact that the cart itself has no "meaningful" identity (it has a workflowId, but what does that means in practice? In a real web app, that could for example relate to a tracking cookie, which we're completely omitting here... And yet, we care about item ids?)

As an alternative, I would use hardcoded values for these ids, rather than randomuuid.

@cretz WDYT?

Reply:
addressed